### ISSUE TITLE
Debian 9, changed behaviour of screen -L

### ISSUE THREAD

**Issue by ragnos** [easy-wi/developer#964]
While testing the upcoming Debian 9 with easy-wi, we noticed a changed behavior of screen's -L argument.

The old behavior of -L was to write it's output into "$(pwd)/screenlog.0", without any further parameters.

The new behavior, however, expects -L to be given a parameter containing the desired filename. That results in parsing the next string after "-L" as filename. This causes the Live-Console not finding the logfile and, more importantly, the gameserver process won't start due to the syntax error at screen.

This change was introduced with screen 4.5.0. Prior to that it was not possible to specify a filename with -L. That means a fix for this issue breaks compatibility to any linux distribution which is using screen <4.5.0

Relevant Code: [stuff/methods/class_app.php Line 1631](https://github.com/easy-wi/developer/blob/9bed858ed8f081a9521a78e1869524c1808685c9/stuff/methods/class_app.php#L1631)

To continue our testing we added a simple hotfix to continue our testing with this particular host:

```php
if ($this->appServerDetails['serverIP'] == '192.168.0.100') {
    $shellCommand .= ' screen -A -m -d -L screenlog.0 -S ' . $this->appServerDetails['serverIP'] . '_' . $this->appServerDetails['port'] . ' ' . $startCommand;
} else {
    $shellCommand .= ' screen -A -m -d -L -S ' . $this->appServerDetails['serverIP'] . '_' . $this->appServerDetails['port'] . ' ' . $startCommand;
}
```

I see two ways of solving this issue: Either check for screen's version within the temporary sh-files and always construct both versions of the commandline, or add another attribute to the gameserver hosts indicating which Linux distribution is used. The latter would create the possibility of distribution-specific fixes in the future.

This issue will reach Ubuntu with it's next release in April.

**Comment by MadMakz** [comment]
Behavior will be changed again in >4.5.0

See http://git.savannah.gnu.org/cgit/screen.git/commit/?h=screen-v4&id=d6c724ad5348c529b87df6fd9be817e1f495451e
http://git.savannah.gnu.org/cgit/screen.git/commit/?h=screen-v4&id=18193bc7b237d66a076fffa21d4308dbb4f83cc5
http://git.savannah.gnu.org/cgit/screen.git/commit/?h=screen-v4&id=13183da34eaf9efb3c6b47371c08ff786eaf712b
http://git.savannah.gnu.org/cgit/screen.git/commit/?h=screen-v4&id=a38de4e6627f01cad43cff57490fd80a988b8e18

That means 4.5.0 is bogous.

**Comment by ragnos** [comment]
Debian 9 has reached it's last stage of freezing before being turned into the new stable release. That means no changes will be made unless it's release critical. And that means this will be the state of screen we will see with Debian 9. After that it's very likely we have to wait until the first point-release of Debian 9 before we see any patch for this, if at all. Note that Debian won't apply all upstream changes, only those the maintainers see necessary or are security-related. And as of now there isn't even a bug-report for this at Debian's Bugtracker. Just to put this into perspective: The screen package of Debian 8 hasn't seen a single patch over the entire lifespan of Jessie apart from one single security-related issue.

Long story short, Debian 9 is coming with a version of screen with a bugged sub-function only a small group of people are actively using. And since you only need a "small change" to make it work again, this won't have top priority at Debian. 

Should this be patched by Debian? Absolutely. Will this be patched in a timely manner? I'm not so sure. Hence one should expect a lot of noise on the forum. Hordes of early-adaptors happily dist-upgrading their servers, only to see easy-wi isn't working anymore. We should at least think about a workaround for this issue for the time being.

**Comment by Lacrimosa99** [comment]
The same is on Ubuntu 17.04..

**Comment by ulrichblock** [comment]
awk to the rescue: `screen -v | awk '{split($3, a, "."); print a[2]}'`
